mainwindow.cpp

```diff
@@ -76,7 +77,7 @@ MainWindow::MainWindow(QWidget *parent)
     /// [F] I understand how it is annoying to not have the database
     /// working when something has changed. Hopefully this check
     /// helps to avoid that in the future!
-    const int DATABASE_REVISION_NUMBER = 14;
+    const int DATABASE_REVISION_NUMBER = 15;
```

Comment:
Instead of making a const variable you should consider doing the following.
```c++
// ADatabase.h
#define DATABASE_VERSION 15
```
That way we can use the this imformatino in a very clean way:
```c++
// any_file_that_interfaces_with_db.h

void function(){
#if DATABASE < 15
    // code for database <15
#else
    // some cool optimisation that is not supported by <15
#endif
}
```
That way the code in the true branch of the preprocessor #if will be put in the file.
And also during developement should there be any hiccup between database revisions we can alleviate it with a preprocessor making it easier than going through the "what is the database version, let me pull it etc etc"

Reply:
Good point, I'll implement it!